### PR TITLE
Introduce `tbot-distroless` image

### LINKED ISSUE #21280: Standalone `tbot` docker image
What would you like Teleport to do?

I would like `teleport` to publish a standalone container image with just the `tbot` binary included, so running it as a sidecar won't required pulling the entire teleport image which is ~600mb in size.

What problem does this solve?

Smaller footprint for running `tbot` as a sidecar container for other applications, smaller image -> less time to pull -> quicker startup, especially in a kubernetes pod sidecar setup.

If a workaround exists, please include it.

Other than building a custom version of the docker image, none.

It might be related to the this thread https://github.com/gravitational/teleport/issues/8735, although specifically I'm referring to `tbot` container image

### PR DESCRIPTION
**Pull request by strideynet**
Closes https://github.com/gravitational/teleport/issues/21280

Paired with https://github.com/gravitational/teleport.e/pull/3465

changelog: tbot-distroless image is now published. This contains just the tbot binary and therefore has a smaller image size.

Successful build: https://drone.platform.teleport.sh/gravitational/teleport/34023
Image: public.ecr.aws/gravitational/tbot-distroless:16.0.0-dev.noah.tbr.1

Working `docker run`:
````
docker run public.ecr.aws/gravitational/tbot-distroless:16.0.0-dev.noah.tbr.1                                                                   
Unable to find image 'public.ecr.aws/gravitational/tbot-distroless:16.0.0-dev.noah.tbr.1' locally
16.0.0-dev.noah.tbr.1: Pulling from gravitational/tbot-distroless
8398841c8311: Pull complete 
2b776ada0341: Pull complete 
2a977872b36c: Pull complete 
fcb6f6d2c998: Pull complete 
e8c73c638ae9: Pull complete 
1e3d9b7d1452: Pull complete 
4aa0ea1413d3: Pull complete 
2fa82a9c76b2: Pull complete 
672354a91bfa: Pull complete 
acd581f1e199: Pull complete 
972a9f56458f: Pull complete 
810b5cc4682a: Pull complete 
45c2519a3853: Pull complete 
2b9c28259b9a: Pull complete 
0d7c3f6d6dc4: Pull complete 
Digest: sha256:1d2d377a93cbff0c66953670c19eef2e793d3c630f361758fec802fe0565078a
Status: Downloaded newer image for public.ecr.aws/gravitational/tbot-distroless:16.0.0-dev.noah.tbr.1
Usage: tbot [<flags>] <command> [<args> ...]

Teleport Machine ID

Machine ID issues and renews short-lived certificates so your machines can
access Teleport protected resources in the same way your engineers do!

Find out more at https://goteleport.com/docs/machine-id/introduction/

Flags:
  -d, --[no-]debug  Verbose logging to stdout.
  -c, --config      Path to a configuration file.
      --[no-]fips   Runs tbot in FIPS compliance mode. This requires the FIPS
                    binary is in use.

Commands:
  help         Show help.
  version      Print the version of your tbot binary.
  start        Starts the renewal bot, writing certificates to the data dir at a set interval.
  init         Initialize a certificate destination directory for writes from a separate bot user.
  configure    Creates a config file based on flags provided, and writes it to stdout or a file (-c <path>).
  migrate      Migrates a config file from an older version to the newest version. Outputs to stdout by default.
  db           Execute database commands through tsh.
  proxy        Start a local TLS proxy via tsh to connect to Teleport in single-port mode.

Try 'tbot help [command]' to get help for a given command.
````

### CODE REVIEW

**Comment by strideynet** [comment]
I'll rebase out the "Release" commit once this is all approved ;)

File: build.assets/charts/Dockerfile-tbot-distroless
Comment (marcoandredinis) [inline build.assets/charts/Dockerfile-tbot-distroless:26]: Would it make sense to wrap `tbot` in `dumb-init` ?

**Review by marcoandredinis — COMMENTED** [review]
This looks good to me, but I would rather have someone from Tooling review this as well

File: build.assets/charts/Dockerfile-tbot-distroless
Comment (strideynet) [reply build.assets/charts/Dockerfile-tbot-distroless:26]: We could but I'm not sure the usual reasons apply - iirc `dumb-init` is necessary when your process doesn't correctly handle signals, but `tbot` explicitly handles these.

**Review by marcoandredinis — APPROVED** [review]

File: examples/chart/teleport-cluster/tests/__snapshot__/auth_deployment_test.yaml.snap
Comment (tigrato) [inline examples/chart/teleport-cluster/tests/__snapshot__/auth_deployment_test.yaml.snap:144]: this is replacing the version with a wrong tag

Comment (strideynet) [reply examples/chart/teleport-cluster/tests/__snapshot__/auth_deployment_test.yaml.snap:144]: Right yeah - I'm going to rebase out the [2a812ee](https://github.com/gravitational/teleport/pull/38259/commits/2a812ee67d50fe2d7aeb0e96de3d1605ed012739) commit once everything is approved and I don't need to cut any more releases :)

File: build.assets/charts/Dockerfile-tbot-distroless
Comment (camscale) [inline build.assets/charts/Dockerfile-tbot-distroless:22]: These commands all seem unnecessary if all you need is the `tbot` binary.

You can also avoid the previous `COPY` command by running

```
RUN --mount=target=/ctx dpkg-deb -R /ctx/$TELEPORT_DEB_FILE_NAME /opt/staging
```

as done in the (now deprecated) teleport image: https://github.com/gravitational/teleport/blob/eb8396b73feb980cff3716c23a5dc5cb4a1f8b4f/build.assets/charts/Dockerfile#L65

File: dronegen/promote.go
Comment (camscale) [inline dronegen/promote.go:46]: Publishing (was "promotion") on master is now fronted by the `tag-publish.yaml` workflow in teleport.e and will no longer be driven by drone. This is currently being backported to the release branches. Could you add the call to your workflow there and not here in drone?

File: dronegen/tag.go
Comment (camscale) [inline dronegen/tag.go:69]: If you don't mind giving this another week or so, I'll have the tag build pipelines off drone completely, real soon now (I hope to start raising the master PRs tomorrow). It would be good to avoid updating drone when it already has the knife in its heart and is slowly bleeding out. If you have a timeline you need to meet for this, let me know and we'll figure something out.

**Review by camscale — COMMENTED** [review]
How quickly do you need to get this in? I'd very much prefer to avoid adding to Drone right now, as I am currently moving everything off it. It's a slow and painful process but I hope to have it complete by the end of the week. If you can hold off a little, I'll work with you to get this into the GHA build/publish workflows instead.

**Comment by strideynet** [comment]
> How quickly do you need to get this in? I'd very much prefer to avoid adding to Drone right now, as I am currently moving everything off it. It's a slow and painful process but I hope to have it complete by the end of the week. If you can hold off a little, I'll work with you to get this into the GHA build/publish workflows instead.

More than happy to hold off - especially if it means this can go straight into GHA instead of drone. I'll regroup with you next week on this - thank you Cam.

**Comment by strideynet** [comment]
Successful build in the pure GHA environment: https://github.com/gravitational/teleport.e/actions/runs/8055743348

**Comment by strideynet** [comment]
Running v16.0.0-dev.noah.tbr.6 build to test [b326428](https://github.com/gravitational/teleport/pull/38259/commits/b326428d67071eb0c35a38ec53d1eaf9ad9150bd)

**Comment by strideynet** [comment]
Once I've got approval on both sides, I'll drop the release commits from this branch.

File: build.assets/charts/Dockerfile-tbot-distroless
Comment (strideynet) [reply build.assets/charts/Dockerfile-tbot-distroless:21]: [b326428](https://github.com/gravitational/teleport/pull/38259/commits/b326428d67071eb0c35a38ec53d1eaf9ad9150bd)

File: build.assets/charts/Dockerfile-tbot-distroless
Comment (camscale) [inline build.assets/charts/Dockerfile-tbot-distroless:21]: It's not clear to me why these commands are needed. The `dpkg-deb` command is enough to extract the `.deb` file, and the executable is copied from this extracted hierarchy. I don't think these commands are doing anything useful.

File: build.assets/charts/Dockerfile-tbot-distroless
Comment (camscale) [inline build.assets/charts/Dockerfile-tbot-distroless:25]: nit: missing eol at eof

**Review by camscale — COMMENTED** [review]
This has still got drone stuff in it that should be dropped. All this PR needs is the `Dockerfile-tbot-distroless` file.

To keep the version stuff from a dev tag build getting on the PR, what I often do is create a test branch on top of the PR branch (`strideynet/test/tbot-docker-image-remastered` in this case) and run `make update-version; make update-tag` on that. The test branch does not even need to be pushed, since the tag is pushed. Then I throw away the branch when I'm done.

I don't like to approve PRs that have the version changes in it in case some accidentally gets merged - I think it's cleaner and easier to leave it off the PR.

**Comment by strideynet** [comment]
Running https://github.com/gravitational/teleport.e/actions/runs/8062741799 to validate deb extraction changes and changes on e to use version action.

**Comment by strideynet** [comment]
```
➜  teleport git:(strideynet/migrate-remote-cluster-rpcs-to-grpc) ✗ docker run public.ecr.aws/gravitational/tbot-distroless:16.0.0-dev.noah.tbr.8
Unable to find image 'public.ecr.aws/gravitational/tbot-distroless:16.0.0-dev.noah.tbr.8' locally
16.0.0-dev.noah.tbr.8: Pulling from gravitational/tbot-distroless
8398841c8311: Pull complete 
2b776ada0341: Pull complete 
2a977872b36c: Pull complete 
fcb6f6d2c998: Pull complete 
e8c73c638ae9: Pull complete 
1e3d9b7d1452: Pull complete 
4aa0ea1413d3: Pull complete 
2fa82a9c76b2: Pull complete 
672354a91bfa: Pull complete 
acd581f1e199: Pull complete 
972a9f56458f: Pull complete 
810b5cc4682a: Pull complete 
45c2519a3853: Pull complete 
2b9c28259b9a: Pull complete 
0be80286843c: Pull complete 
Digest: sha256:15008c9ada589182e39aea6914dc53d0bc409ab46cb8e0eea5b9b284b9b2f32b
Status: Downloaded newer image for public.ecr.aws/gravitational/tbot-distroless:16.0.0-dev.noah.tbr.8
Usage: tbot [<flags>] <command> [<args> ...]

Teleport Machine ID

Machine ID issues and renews short-lived certificates so your machines can
access Teleport protected resources in the same way your engineers do!

Find out more at https://goteleport.com/docs/machine-id/introduction/

Flags:
  -d, --[no-]debug  Verbose logging to stdout.
  -c, --config      Path to a configuration file.
      --[no-]fips   Runs tbot in FIPS compliance mode. This requires the FIPS
                    binary is in use.

Commands:
  help         Show help.
  version      Print the version of your tbot binary.
  start        Starts the renewal bot, writing certificates to the data dir at a set interval.
  init         Initialize a certificate destination directory for writes from a separate bot user.
  configure    Creates a config file based on flags provided, and writes it to stdout or a file (-c <path>).
  migrate      Migrates a config file from an older version to the newest version. Outputs to stdout by default.
  db           Execute database commands through tsh.
  proxy        Start a local TLS proxy via tsh to connect to Teleport in single-port mode.

Try 'tbot help [command]' to get help for a given command.
```

Successful build, publish and run based on the last commit!

**Review by camscale — APPROVED** [review]
Looks really good now. I'm pleased with how simple it's ended up.

**Review by tigrato — APPROVED** [review]

**Pull request merged**
